### ISSUE TITLE
odd behaviour with useSSQ and suspense boundary

### ISSUE THREAD

**Issue by tigawanna** [rakkasjs/rakkasjs#186]
Just jumped on the new release and it's awesome, Am using the latest next.30
i have an simple postgres gui that's supposed to open a db and return all the tables  from a list of all avaialbel dbs 
am using useSSQ to run it node side 
```tsx
  const query = useSSQ(async (ctx) => {
    console.log("This is a one database ", db_name);
    try {
      const config = safeDestr<DbAuthProps>(ctx.cookie?.pg_cookie);
      const sql = postgresInstance(config);
      const tables = (await sql`
        SELECT table_name,
    (SELECT string_agg(column_name, ', ')
    FROM information_schema.columns
    WHERE table_schema = t.table_schema
        AND table_name = t.table_name) AS columns,
    (SELECT string_agg(data_type, ', ')
    FROM information_schema.columns
    WHERE table_schema = t.table_schema
        AND table_name = t.table_name) AS column_types
    FROM information_schema.tables t
  WHERE table_schema = 'public';

        `) as any as [
        {
          table_name: string;
          columns: string;
          column_types: string;
        },
      ];
      console.log(" === tables == ", tables[0]);
      return { tables, error: null };
    } catch (error: any) {
      console.log(" === error == ", error.message);
      return { tables: null, error: error.message };
    }
  });
```

Am noticing odd behaviour where it seems like useSSQ is holding on to the firsts data fetched until the page is refreshed , locally i have a sherpa and dvdrental db and some other default ones , when i open dvdrental and go back and open any other database i'll keep seeing the dvd ental results , mind you i save the current database and it's passowrd in a cookie which gets updated every time i navigate to a new DB 
i've noticed that adding a cosole.log on top of the try catch statement in the useSSQ fixes the issue 

[fetching-issue.webm](https://github.com/rakkasjs/rakkasjs/assets/72096712/ce00e8ef-516c-498b-8a1d-3c841e7eddf2)

there's also an issue with suspense fallbacks not trigering and something happenig to css where all the padding is lost when it's suspending , i've noticed the intended  suspese fallback getting used sometimes but i can't rectreate it reliably

[suspense-issue.webm](https://github.com/rakkasjs/rakkasjs/assets/72096712/c7fed96d-f108-479c-81da-7782b06a5ecb)

The rest of the code is [here](https://github.com/tigawanna/postbase) in case you need to look at it

**Comment by cyco130** [comment]
Hi!

## `useSSQ` keeping the old data

Behind the scenes, `useQuery` and `useSSQ` use a cache with the query key as the cache key. With `useQuery`, the query key is explicit. You can also provide an explicit key to `useSSQ` using the second parameter but if you don't, Rakkas tries to create one using captured values used in the query function. When using cookie for the database name, Rakkas has no way of knowing that the data dependencies have changed. That's why adding `console.log` fixes the issue, now Rakkas nows that the query depends on `db_name`. To fix, you can either use `db_name` instead of the cookie or provide an explicit key that contains it, e.g. something like:

```ts
useSSQ(
  async () => {
    // ...
  },
  { key: `db/${db_name}` }
);
```

## Suspense boundaries

I'm not sure I understand the issues with suspense boundaries exactly but Rakkas's client router uses a React "transition" that waits for the outermost suspense boundaries to resolve before switching so that the user keeps seeing the old view instead of seeing a loading state. So those boundaries will only be seen when doing a hard load. I believe you could add one more boundary to force showing it but I'm not sure.

## CSS kicking late

Is this dev or prod? If it's dev, it might be about Rakkas not understanding how Tailwind injects styles. I'll do a few tests but it would help a lot if you could provide a small repro.

**Comment by tigawanna** [comment]
Adding keys fixed the issue and the rest went away after bulding , tried to re-create the funny css in dev mode and i can't , tailwind and regular css work just fine in a brand new project

**Comment by cyco130** [comment]
Good to hear!

I'll keep this open until I do my own Tailwind experiments. Maybe I'll get lucky and reproduce the behavior. I have a hunch that it involves `React.lazy` or similar `Suspense` stuff.

**Comment by cyco130** [comment]
Are you importing your global CSS file from `client-entry` by any chance?

If so, #197 will fix it. I'm hoping to release today but you can import your global CSS file from your main layout as a workaround in the meantime.

**Comment by tigawanna** [comment]
am not using client-entry the styles are being imported in the root layout

**Comment by cyco130** [comment]
It might still fix it, because I've fixed several other prefetch injection issues. There are two more bugs I want to address before releasing, then we can check :)

**Comment by cyco130** [comment]
It was a CSS escaping bug. Should be fixed in `0.7.0-next.35` (currently being published).
